dbt/include/athena/macros/materializations/models/table/table.sql

```diff
@@ -12,19 +12,46 @@
 
   {{ run_hooks(pre_hooks) }}
 
-  -- cleanup
-  {%- if old_relation is not none -%}
-    {{ drop_relation(old_relation) }}
-  {%- endif -%}
+  {%- if old_relation is none or table_type != 'iceberg' -%}
+    {%- if old_relation is not none -%}
+      {{ drop_relation(old_relation) }}
+    {%- endif -%}
 
-  -- build model
-  {% call statement('main') -%}
-    {{ create_table_as(False, target_relation, sql) }}
-  {%- endcall %}
+    {%- call statement('main') -%}
+      {{ create_table_as(False, target_relation, sql) }}
+    {%- endcall %}
 
-  {% if table_type != 'iceberg' %}
-    {{ set_table_classification(target_relation) }}
-  {% endif %}
+    {%- if table_type != 'iceberg' -%}
+      {{ set_table_classification(target_relation) }}
+    {%- endif -%}
+  {%- else -%}
+    {%- set tmp_relation = api.Relation.create(identifier=target_relation.identifier ~ '__ha',
+                                               schema=schema,
+                                               database=database,
+                                               s3_path_table_part=target_relation.identifier,
```

Comment:
nice job with the s3_path_table_part, we should refactor the table_hive_ha to use that too, to be consistent.